runtime/src/main/java/org/corfudb/runtime/collections/StreamSubscription.java

```diff
@@ -137,7 +138,12 @@ boolean enqueueStreamEntry(ILogData logData, long maxBlockTime) throws Interrupt
             return true;
         }
 
-        return streamBuffer.offer(new CorfuStreamEntries(streamEntries), maxBlockTime, TimeUnit.NANOSECONDS);
+        Timestamp timestamp = Timestamp.newBuilder()
+                .setSequence(smrEntries.getGlobalAddress())
```

Comment:
Since the epoch is derived from `logData` can you also use it for `gobalAddress`. The order of when `setGlobalAddress` is called on the payload is not clear.

Reply:
It should be the same, but yes no problem. Done.